pyuvdata/uvdata/tests/test_miriad.py

```diff
@@ -192,6 +192,68 @@ def test_read_write_read_carma(tmp_path):
     assert uv_in == uv_out
 
 
+@pytest.mark.filterwarnings("ignore:LST values stored in this file are not ")
+@pytest.mark.filterwarnings("ignore:The uvw_array does not match the expected values")
+def test_read_carma_miriad_write_ms(tmp_path):
+    pytest.importorskip("casacore")
+    uv_in = UVData()
+    uv_out = UVData()
+    carma_file = os.path.join(DATA_PATH, "carma_miriad")
+    testfile = os.path.join(tmp_path, "outtest_carma_miriad.ms")
+
+    with uvtest.check_warnings(
+        UserWarning,
+        [
+            "Altitude is not present in Miriad file, "
+            "using known location values for SZA.",
+            "The uvw_array does not match the expected values given the antenna "
+            "positions.",
+            "pamatten in extra_keywords is a list, array or dict",
+            "psys in extra_keywords is a list, array or dict",
+            "psysattn in extra_keywords is a list, array or dict",
+            "ambpsys in extra_keywords is a list, array or dict",
+            "bfmask in extra_keywords is a list, array or dict",
+            "Cannot fix the phases of multi phase center datasets, as they were not "
+            "supported when the old phasing method was used, and thus, there "
+            "is no need to correct the data.",
+        ],
+    ):
+        uv_in.read(carma_file, fix_old_proj=True)
+
+    # Dump the extra keywords here, since they aren't written to MS format anyways
+    uv_in.extra_keywords = {}
```

Comment:
Is there not a way to write generic header keywords to ms files?

Reply:
Not that I can easily see -- one might be able to add arbitrary keywords to the full table itself (and I can try to experiment with that a bit), but I have no idea how CASA will behave with those in there. Let me give it a try and see what happens...

Reply:
Okay, I take back my previous comments. Arbitrary generic header keywords for everyone!

Reply:
Alright, added in 3068e067, the full dict in `extra_keywords` is now stored under the main MS table under the name `pyuvdata_extra`.